Mathlib/Order/Directed.lean

```diff
@@ -321,3 +321,65 @@ instance (priority := 100) OrderTop.to_isDirected_le [LE α] [OrderTop α] : IsD
 -- see Note [lower instance priority]
 instance (priority := 100) OrderBot.to_isDirected_ge [LE α] [OrderBot α] : IsDirected α (· ≥ ·) :=
   ⟨fun _ _ => ⟨⊥, bot_le _, bot_le _⟩⟩
+
+namespace DirectedOn
+
+section Pi
+
+namespace Pi
+
+variable {ι : Type*} {α : ι → Type*} [∀ i, LE (α i)]
+
+lemma proj {d : Set (Π i, α i)} (hd : DirectedOn (· ≤ ·) d) (i : ι) :
+    DirectedOn (· ≤ ·) ((fun a => a i) '' d) := by
+  intro p ⟨a, ha⟩ q ⟨b, hb⟩
+  obtain ⟨z,hz⟩ := hd a ha.1 b hb.1
+  use z i
+  aesop
+
+lemma prodMk {d : (i : ι) → Set (α i)} (hd : ∀ (i : ι), DirectedOn (· ≤ ·) (d i)) :
+    DirectedOn (· ≤ ·) (Set.pi  Set.univ d) := by
```

Comment:
Could you allow for any `s : Set ι` instead of `univ`? 

Also, I would call this `DirectedOn.set_pi`, or even `DirectedOn.pi`

Reply:
@ADedecker I'm failing to imagine the statement you want proved here? I assume you mean something like `DirectedOn (· ≤ ·) (Set.pi  s d)` - but given x and y in `Set.pi  s d` I think one still needs to find `z`  in `Set.pi  s d` such that `x i ≤ z i` and `y i ≤ z i` for all `i` in `ι`, not just in `s`?

Reply:
You're right, this was a dumb suggestion. The worst part is I remember "thinking" about it, but clearly I just made something up. Sorry for that.

Reply:
Easily done! Particularly when switching between lean programming and mathematical thinking.